packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -81,13 +45,43 @@ export interface QueryParams {
     shortCode?: string
 }
 
-export interface QueryResponse<T> extends CommonHookResponse {
-    data: T
+export type QueryResponse<T> =
+    | {
+          isLoading: boolean
+          error?: undefined
+          data?: undefined
+      }
+    | {
+          isLoading: true
+          error?: Error
+          data?: T
+      }
+    | {
+          isLoading: false
+          error: Error
+          data?: undefined
+      }
+    | {
+          isLoading: false
+          error?: undefined
+          data: T
+      }
```

Comment:
Dumb question, Can we do something like this? Why do we have to separate it out? 
```
{
 isLoading: boolean
error? : undefined | Error 
data: T | undefined 
}
```

Reply:
That type is valid, but it's a bit broader than necessary. For example, that would allow `{ isLoading: true, error: Error, data: T }`, but we know we'll never actually see that scenario (`error` and `data` will both only be set if `isLoading` is false, and we will only have one of the two, not both). Since this interface is pretty small, it's pretty simple to just enumerate the valid scenarios.